starter/privileged_service_client.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#if !defined(__linux__)
+#error "Linux platform file is part of non-Linux build."
+#endif
+
+#include "source/common/common/assert.h"
+#include "source/common/singleton/threadsafe_singleton.h"
+
+#include "starter/privileged_service_protocol.h"
+
+namespace Envoy {
+namespace Cilium {
+
+class Bpf;
+class SocketMarkOption;
+
+namespace PrivilegedService {
+
+// ProtocolClient implements the client logic for communicating with the privileged service.
+class ProtocolClient : public Protocol {
+public:
+  ProtocolClient();
+
+  // allow access to the classes that need it
+  friend class Envoy::Cilium::Bpf;
+  friend class Envoy::Cilium::SocketMarkOption;
+
+protected:
+  // Read-only bpf syscalls
+  Envoy::Api::SysCallIntResult bpf_open(const char *path);
+  Envoy::Api::SysCallIntResult bpf_lookup(int fd, const void *key, uint32_t key_size, void* value,
+					  uint32_t value_size);
+
+  // Set a socket option
+  Envoy::Api::SysCallIntResult setsockopt(int sockfd, int level, int optname, const void *optval,
+					  socklen_t optlen);
+
+private:
+  bool check_privileged_service();
+  bool running_with_cilium_privileged_service() const { return is_open(); }
```

Comment:
rename to `connected_to_cilium_privileged_service` ?

Reply:
IMO this was (already) getting too long, renamed as `have_cilium_privileged_service()`